assets/src/design-system/components/dropDown/menu/listGroupings.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+
+import { getInset } from '../utils';
+import { DROP_DOWN_VALUE_TYPE, MENU_OPTIONS } from '../types';
+import DefaultListItem from './defaultListItem';
+import List from './list';
+
+const ListGroupings = ({
+  options,
+  activeValue,
+  listLength,
+  listId,
+  hasMenuRole,
+  handleMenuItemSelect,
+  renderItem,
+  optionsRef,
+}) => {
+  const ListItem = renderItem || DefaultListItem;
+  const isManyGroups = useMemo(() => options?.length > 1, [options.length]);
```

Comment:
Since this is just testing if something is > than a value, it doesn't need to be memoized. The comparison could be more expensive than the test. Alternatively if this only changes once within the lifecycle of the ListGrouping, we could put this in a `useRef` like `const isManyGroups = useRef(Boolean(options?.length > 1)).current`